lib/systems/default.nix

```diff
@@ -384,7 +384,11 @@ let
             }.${cpu.name} or cpu.name;
             vendor_ = final.rust.platform.vendor;
           # TODO: deprecate args.rustc in favour of args.rust after 23.05 is EOL.
-          in args.rust.rustcTarget or args.rustc.config
+          in
+          # Rust uses `wasm32-wasi` rather than `wasm32-unknown-wasi`.
+          if final.isWasi
+          then "${cpu_}-wasi"
```

Comment:
This is in the wrong place, as it currently takes precedence over explicit overrides from the user.  It should be after the `or`.

Reply:
Fixed.